src/pymor/models/iosys.py

```diff
@@ -675,55 +675,59 @@ def to_abcde_files(self, files_basename, mu=None):
-        if output:
-            D = LinearInputOperator(self.D)
-            data['output'] = np.empty((n, self.dim_output))
-            data_output_extra = []
-        for i, (x, t) in enumerate(iterator):
-            if solution:
-                data['solution'].append(x)
-            if output:
-                y = self.C.apply(x, mu=mu).to_numpy() + D.as_range_array(mu=mu.with_(t=t)).to_numpy()
-                if i < n:
-                    data['output'][i] = y
-                else:
-                    data_output_extra.append(y)
-        if output:
-            if data_output_extra:
-                data['output'] = np.vstack((data['output'], data_output_extra))
-            if len(data['output']) < i + 1:
-                data['output'] = data['output'][:i + 1]
-        return data
+    def _compute(self, quantities, data, mu):
+        if 'solution' in quantities or 'output' in quantities:
+            assert self.T is not None
+
+            compute_solution = 'solution' in quantities
+            compute_output = 'output' in quantities
+
+            # solution computation
+            iterator = self.time_stepper.iterate(
+                0,  # initial_time
+                self.T,  # end_time
+                self.initial_data.as_range_array(mu),  # initial_data
+                -self.A,  # operator
+                rhs=LinearInputOperator(self.B),
+                mass=None if isinstance(self.E, IdentityOperator) else self.E,
+                mu=mu.with_(t=0),
+                num_values=self.num_values
+            )
+            if self.num_values is None:
+                try:
+                    n = self.time_stepper.estimate_time_step_count(0, self.T) + 1
+                except NotImplementedError:
+                    n = 0
+            else:
+                n = self.num_values + 1
+
+            if compute_solution:
+                data['solution'] = self.solution_space.empty(reserve=n)
+            if compute_output:
+                D = LinearInputOperator(self.D)
+                data['output'] = np.empty((n, self.dim_output))
+                data_output_extra = []
+            for i, (x, t) in enumerate(iterator):
+                if compute_solution:
+                    data['solution'].append(x)
+                if compute_output:
+                    y = self.C.apply(x, mu=mu).to_numpy() + D.as_range_array(mu=mu.with_(t=t)).to_numpy()
+                    if i < n:
+                        data['output'][i] = y
+                    else:
+                        data_output_extra.append(y)
+            if compute_output:
+                if data_output_extra:
+                    data['output'] = np.vstack((data['output'], data_output_extra))
+                if len(data['output']) < i + 1:
+                    data['output'] = data['output'][:i + 1]
+
+            if compute_solution:
+                quantities.remove('solution')
+            if compute_output:
+                quantities.remove('output')
```

Comment:
```suggestion
            if compute_solution:
                data['solution'] = self.solution_space.empty(reserve=n)
                time_instances = []
                for i, (x, t) in enumerate(iterator):
                    data['solution'].append(x)
                    time_instances.append(t)
                quantities.remove('solution')
            if compute_output:
                D = LinearInputOperator(self.D)
                data['output'] = np.empty((n, self.dim_output))
                data_output_extra = []
                for i, (x, t) in enumerate(zip(data['solution'], time_instances)):
                    y = self.C.apply(x, mu=mu).to_numpy() + D.as_range_array(mu=mu.with_(t=t)).to_numpy()
                    if i < n:
                        data['output'][i] = y
                    else:
                        data_output_extra.append(y)
                if data_output_extra:
                    data['output'] = np.vstack((data['output'], data_output_extra))
                if len(data['output']) < i + 1:
                    data['output'] = data['output'][:i + 1]
                quantities.remove('output')
```

Just as an idea for how to get rid of all the `if compute_...`.

Reply:
Maybe I misunderstand your suggestion, but the point of the original code is that when you are only interested in the output you don't need to keep the entire state space trajectory in memory ..

Reply:
Ah yes, of course. You are right!